composer/trainer/trainer.py

```diff
@@ -1652,15 +1663,21 @@ def _train_batch(self, use_grad_scaling: bool):
                             total_loss = self.state.scaler.step(
                                 optimizer, closure=lambda **kwargs: self._train_microbatches(microbatches, **kwargs))
                         else:
-                            total_loss = optimizer.step(
-                                closure=lambda **kwargs: self._train_microbatches(microbatches, **kwargs).item())
+                            if type(self._device) == DeviceTPU:
```

Comment:
nit: prefer `isinstance(..., xxx)` rather than `type(...) is xxx` as the former allows for subclassing 

```suggestion
                            if isinstance(self._device, DeviceTPU):
```